working_villagers/api.lua

```diff
@@ -845,7 +845,13 @@ function working_villages.register_villager(product_name, def)
       self.product_name = name
       self.manufacturing_number = working_villages.manufacturing_data[name]
       working_villages.manufacturing_data[name] = working_villages.manufacturing_data[name] + 1
-      create_inventory(self)
+      local inventory = create_inventory(self)
+
+      if self.object:get_luaentity().owner_name == "working_villages:self_employed" then
+        -- Spawned in with no owner, assign job
+        local job_stack = ItemStack("working_villages:job_herbcollector")
+        inventory:set_stack("job", 1, job_stack)
+      end
```

Comment:
I think, job assignments should be part of the spawn function.
It allows you to spawn other types of self-employed villagers in the future.

Reply:
The technical challenge is that Minetest calls the new villager's on_activate() function after the spawning function exits, and that is a prerequisite to being able to assign the job. I can look for another way to pass the requested job.

One possibility is to set the job name as the owner_name at spawn time, and change the owner and set the specified job in on_activate.

Reply:
I understand.

What something like this:
```lua
      if self.object:get_luaentity().__self_employed_job then
        -- Spawned in with no owner, assign job
        local job_stack = ItemStack("working_villages:"..self.object:get_luaentity().__self_employed_job)
        inventory:set_stack("job", 1, job_stack)
      end
```

In spawn function add:
```lua
new_villager:get_luaentity().__self_employed_job = "job_herbcollector"
```

Reply:
Further testing revealed that sometimes on_activate() ran before the code following add_entity(), so I had to move the logic around a fair bit, but it works well now. I also added an option to avoid spawning near owned objects, to prevent the villagers
from griefing my friend's island.... Probably need to integrate with a better protection mechanism (later).